viewing_party/party.py

```diff
@@ -1,23 +1,186 @@
+            highest_count = count
+            highest_genre = genre
+
+    #print(highest_genre)
+    return highest_genre
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
+def get_unique_watched(user_data):
+    #Assign empty list to variables
+    user_unique_watched = []
+    friend_movie_title = []
+
+    #loop through wantched list from friend
+    for friend in user_data["friends"]:
+        friend_list = friend["watched"]
+        for movie in friend_list:
+            #add movie title into friend_movie_title list
+            friend_movie_title.append(movie["title"])
+
+    #loop through user's watched list 
+    for user in user_data["watched"]:
+        # if user's watched list movie not in friend's list then add it to the user_unique_watched list
+        if user["title"] not in friend_movie_title:
+            user_unique_watched.append(user)
+    return user_unique_watched
+
+
+def get_friends_unique_watched(user_data):
+    #Assign empty list to variables
+    user_movie_title = []
+    friend_unique_watched = []
+    f_uni_watched_rm_dup = []
+
+    for user in user_data["watched"]:
+        user_movie_title.append(user["title"])
+
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            #remove duplicate titles          
+            if movie["title"] not in user_movie_title \
+                and movie["title"] not in friend_unique_watched:
+                f_uni_watched_rm_dup.append(movie)
+                friend_unique_watched.append(movie["title"])
+
+    return f_uni_watched_rm_dup
+
 
-        
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(user_data):
+    friend_unique_movies = get_friends_unique_watched(user_data)
+
+    # Recommend only movies available on user’s subscriptions
+    recommended = [
+        movie for movie in friend_unique_movies
+        if movie["host"] in user_data["subscriptions"]
+    ]
+
+    #print(recommended)
+    return recommended
+
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
+def get_new_rec_by_genre(user_data):
+    #grab user_movie_list, friend_movie_list, and user_genre
+    user_watched = get_unique_watched(user_data)
+    friend_watched = get_friends_unique_watched(user_data)
+    most_f_genre = get_most_watched_genre(user_data)
+
+    genre_rec_movie = []
+
+    for friend_movie in friend_watched:
+        if friend_movie["genre"] == most_f_genre:
+            genre_rec_movie.append(friend_movie)
+
+    return genre_rec_movie
+
+
+
+def get_rec_from_favorites(user_data):
+    user_unique_watch = get_unique_watched(user_data)
+
+    user_unique_watch_title = set(movie["title"] for movie in user_unique_watch)

```

Comment:
If we're making a `set` from a comprehension, we can use comprehension syntax directly with set literal syntax.
```py
    user_unique_watch_title = {movie["title"] for movie in user_unique_watch}
```

as it was written, Python makes something called a "generator", which runs a little less efficiently than using the set literal syntax.